Makefile

```diff
@@ -21,11 +21,11 @@ CPPFLAGS += -DNDEBUG -D_REENTRANT					\
 ifeq ($(BUILDOS),Darwin)
 #	As of macOS 12, we can't use the system python anymore, so take the
 #	first one that's not in /usr/bin.
-	PY_EXEC=$(shell which -a python2.7 | grep -v /usr/bin/python | head -1)
+	PY_EXEC=$(shell type -a --path python2.7 | grep -v /usr/bin/python | head -1)
 else
-	PY_EXEC=$(shell which python2.7)
+	PY_EXEC=$(shell /usr/bin/command -v python2.7)
```

Comment:
THIS is the magic that works on both bash and ksh93.  Thank you, and thank @jperkin too!